.github/workflows/on-pr.yaml

```diff
@@ -0,0 +1,54 @@
+name: Test CI/CD Pipeline
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+
+env:
+  CI: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: ["20.19.4"]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
```

Comment:
If we end up dropping the matrix (discussed in another comment), we could use the `node-version-file` parameter of this action to make it look at our `.nvmrc` file to get the node version.